velox/docs/functions/presto/conversion.rst

```diff
@@ -495,6 +520,61 @@ Valid examples
   SELECT cast(date '1970-01-01' as timestamp); -- 1970-01-01 00:00:00
   SELECT cast(date '2012-03-09' as timestamp); -- 2012-03-09 00:00:00
 
+From timestamp with time zone
+^^^^^^^^^
+
+Casting from timestamp with time zone to timestamp is allowed. If configuration property
+`adjust_timestamp_to_session_timezone` is set to true, the timestamp will be treated as
+UTC with time zone simply being dropped; otherwise the timestamp will be shifted by the
+value of the offset of the given time zone. For instance: if the option is true, then
+timestamp "1970-01-01 00:00:00.000 America/Los_Angeles" would be adjusted to
+"1970-01-01 00:00:00.000"; otherwise, it would be "1969-12-31 16:00:00.000".
+
+Valid examples
+
+::
+
+  -- `adjust_timestamp_to_session_timezone` is true
+  SELECT cast(timestamp '1970-01-01 00:00:00 America/Los_Angeles' as timestamp); -- 1970-01-01 00:00:00.000
+  SELECT cast(timestamp '2012-03-09 10:00:00 Asia/Chongqing' as timestamp); -- 2012-03-09 10:00:00.000
+  SELECT cast(from_unixtime(0, '+06:00') as timestamp); -- 1970-01-01 00:00:00.000
+  SELECT cast(from_unixtime(0, '-02:00') as timestamp); -- 1970-01-01 00:00:00.000
+
+  -- `adjust_timestamp_to_session_timezone` is false
+  SELECT cast(timestamp '1970-01-01 00:00:00 America/Los_Angeles' as timestamp); -- 1969-12-31 16:00:00.000
+  SELECT cast(timestamp '2012-03-09 10:00:00 Asia/Chongqing' as timestamp); -- 2012-03-09 18:00:00.000
+  SELECT cast(from_unixtime(0, '+06:00') as timestamp); -- 1970-01-01 06:00:00.000
+  SELECT cast(from_unixtime(0, '-02:00') as timestamp); -- 1969-12-31 22:00:00.000
+
+Cast to Timestamp with Time Zone
+-----------------
+
+From timestamp
+^^^^^^^^^^^^
+
+Casting from timestamp to timestamp with time zone is allowed. If configuration property
```

Comment:
Let's rewrite a bit for clarity.

```
The results depend on whether configuration property `adjust_timestamp_to_session_timezone` is set or not.

If set, ...

Otherwise, ...
```